pipeline/inferrer/inference_manager/src/main/scala/uk/ac/wellcome/platform/inference_manager/services/InferrerAdapter.scala

```diff
@@ -0,0 +1,90 @@
+package uk.ac.wellcome.platform.inference_manager.services
+
+import akka.http.scaladsl.model._
+import akka.http.scaladsl.unmarshalling.Unmarshal
+import akka.stream.Materializer
+import de.heikoseeberger.akkahttpcirce.FailFastCirceSupport._
+import grizzled.slf4j.Logging
+import io.circe.Decoder
+import io.circe.generic.semiauto.deriveDecoder
+import uk.ac.wellcome.models.work.internal.{
+  AugmentedImage,
+  InferredData,
+  MergedImage,
+  Minted
+}
+import uk.ac.wellcome.platform.inference_manager.models.FeatureVectorInferrerResponse
+
+import scala.concurrent.Future
+
+/*
+ * An InferrerAdapter is specific to the inferrer and the data that is being augmented.
+ * Implementors must provide:
+ * - The type of the inferrer response
+ * - A Decoder for that response
+ * - A function to create an HTTP request from the input data type
+ * - A function to augment input data with a response, returning the output data type
+ *
+ * Additionally, the trait provides the logic for handling different HTTP response statuses
+ */
+
+trait InferrerAdapter[Input, Output] extends Logging {
+  type InferrerResponse
+
+  implicit val responseDecoder: Decoder[InferrerResponse]
+  def createRequest(input: Input): HttpRequest
+  def augmentInput(input: Input,
+                   inferrerResponse: Option[InferrerResponse]): Output
+
+  def parseResponse(response: HttpResponse)(
+    implicit mat: Materializer): Future[Option[InferrerResponse]] =
+    response.status match {
+      case StatusCodes.OK =>
+        Unmarshal(response.entity).to[Some[InferrerResponse]]
+      case StatusCodes.BadRequest =>
+        Future.failed(new Exception("Bad request"))
+      case StatusCodes.NotFound =>
+        Future.failed(new Exception("Entity not found"))
+      case statusCode =>
+        warn(
+          s"Request failed non-deterministically with code ${statusCode.value}")
+        Future.successful(None)
+    }
+}
+
+// The InferrerAdaptor for feature vectors, consuming MergedImages and
+// augmenting them into AugmentedImages
+object FeatureVectorInferrerAdapter
+    extends InferrerAdapter[MergedImage[Minted], AugmentedImage[Minted]] {
+  type InferrerResponse = FeatureVectorInferrerResponse
+
+  def createRequest(image: MergedImage[Minted]): HttpRequest =
+    HttpRequest(
+      method = HttpMethods.GET,
+      uri = Uri("/feature-vectors/").withQuery(
```

Comment:
I'm not sure if this not implemented fully, or I'm not getting it, but where does this API live?

Reply:
Have a look in here for the design https://github.com/wellcomecollection/docs/tree/master/rfcs/021-data_science_in_the_pipeline - it doesn't exist in this repo yet

Reply:
So the plan is get the inference manager into the pipeline now and passthrough messages (as nothing will be interesting to it yet)?

Reply:
I hadn't actually thought of that! Maybe, but probably not - I haven't done the infra for it yet nor have we integrated the inferrer (although that is done in the data-science repo now). We'll see what order the work gets done in, I guess.

Reply:
Ah, so it's going to be an API that does magic? Cool.

Reply:
We should try to organise a situation where we can deploy this service as soon as possible into a production environment even if it does nothing so we can kind of soak test it, and of course it makes it will make it easier to add the actual functionality later on.